work_queue/src/work_queue_worker.c

```diff
@@ -442,6 +451,30 @@ static void report_worker_ready( struct link *master )
 	send_master_message(master,"workqueue %d %s %s %s %d.%d.%d\n",WORK_QUEUE_PROTOCOL_VERSION,hostname,os_name,arch_name,CCTOOLS_VERSION_MAJOR,CCTOOLS_VERSION_MINOR,CCTOOLS_VERSION_MICRO);
 	send_master_message(master, "info worker-id %s\n", worker_id);
 	send_features(master);
+
+	//attempt to find local TLQ server to retrieve master URL
+	if(tlq_port && debug_path && !tlq_url) {
+		debug(D_TLQ, "looking up worker TLQ URL");
+		tlq_url = address_get_tlq_url(tlq_port, debug_path);
+		if(tlq_url) debug(D_TLQ, "set worker TLQ URL: %s", tlq_url);
+		else {
+			debug(D_TLQ, "error setting worker TLQ URL - setting it to NONE");
+			tlq_url = "NONE";
+		}
+	}
+
+	if(tlq_url) {
```

Comment:
I would avoid blocking for a response here -- I don't think there is any other case where a blocking RPC is initiated by the worker, and strange things will happen if you have a version message.

Instead, move this logic down into `handle_master` so you can simple handle a `tlq` message whenever it happens to arrive.